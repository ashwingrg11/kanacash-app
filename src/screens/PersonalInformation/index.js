import React from 'react';
import {StyleSheet} from 'react-native';
import GenericView from '~/components/ui/GenericView';
import {MediumText} from '~/components/ui/Text';
import Button from '~/components/ui/Button';
import Header from '~/components/ui/Header';
import {ProfileCard} from './container';
import {getCurrentUser} from '~/store/actions/UserDetailsAction';
import {useDispatch, useSelector} from 'react-redux';
import widgetType from '~/constants/widgetType';
import {
  NAVIGTION_TO_WIDGETS_SCREEN,
  NAVIGATION_TO_CREATE_BENEFICIARY_SCREEN,
  NAVIGATION_TO_TIER_SCREEN,
} from '../../navigation/routes';
import {useFocusEffect} from '@react-navigation/native';
import STATUS from '~/constants/widgetStatus';
import {showToast} from '~/store/actions/ToastAction';

export default function PersonalInformation({navigation}) {
  const dispatch = useDispatch();
  const userData = useSelector(state => state.auth);

  useFocusEffect(
    React.useCallback(() => {
      const unsubscribe = dispatch(getCurrentUser());
      return () => unsubscribe;
    }, []),
  );

  const onPressVerifyAccount = () => {
    if (STATUS.REVIEW_PENDING === userData?.status?.kycStatus) {
      dispatch(
        showToast({
          message: 'Your information verification is in progress.',
          status: false,
        }),
      );
    } else {
      navigation.navigate(NAVIGTION_TO_WIDGETS_SCREEN, {
        widgetType: widgetType.kyc,
      });
    }
  };

  return (
    <GenericView padding isScrollable header={<Header title="Profile" />}>
      <ProfileCard
        onPressVerify={onPressVerifyAccount}
        data={userData.user}
        status={userData.status}
        isVerified={userData.status.isKYCVerified}
      />
      <MediumText
        text="To send more money, upgrade your transaction limit"
        style={styles.marginBottom}
      />
      {userData.status.isKYCVerified && (
        <Button
          text="Upgrade Limit"
          style={styles.marginBottom}
          onPress={() =>
            navigation.navigate('Tier', {screen: NAVIGATION_TO_TIER_SCREEN})
          }
        />
      )}
      <Button
        text="Add the Beneficiary"
        style={styles.marginBottom}
        onPress={() =>
          navigation.navigate(NAVIGATION_TO_CREATE_BENEFICIARY_SCREEN, {
            update: false,
          })
        }
      />
      {!userData?.status.isKYCVerified && (
        <Button
          text="Verify your Account"
          buttonInvert
          onPress={onPressVerifyAccount}
          style={styles.marginBottom}
        />
      )}
    </GenericView>
  );
}

const styles = StyleSheet.create({
  marginBottom: {
    marginBottom: 20,
  },
});
