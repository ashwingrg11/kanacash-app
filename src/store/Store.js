import {applyMiddleware, createStore} from 'redux';
import {persistReducer, persistStore} from 'redux-persist';
import thunk from 'redux-thunk';
import rootReducer from './rootReducer';
import {composeWithDevTools} from 'redux-devtools-extension';
import AsyncStorage from '@react-native-community/async-storage';

const config = {
  key: 'root',
  storage: AsyncStorage,
  whitelist: [],
  blackList: [],
};

const reducer = persistReducer(config, rootReducer);
const middleware = composeWithDevTools(applyMiddleware(thunk));

export default data => {
  const store = createStore(reducer, middleware);
  const persistor = persistStore(store);
  return {persistor, store};
};
