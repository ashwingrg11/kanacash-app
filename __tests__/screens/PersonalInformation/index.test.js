import 'react-native';
import React from 'react';

import {SafeAreaProvider} from 'react-native-safe-area-context';

import PersonalInformation from '~/screens/PersonalInformation';
import Store from '~/store/Store';

// Note: test renderer must be required after react-native.
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

describe('<PersonalInformation/>', () => {
  it('PersonalInformation renders correctly', () => {
    const tree = renderer
      .create(
        <SafeAreaProvider store={Store}>
          <PersonalInformation />,
        </SafeAreaProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
