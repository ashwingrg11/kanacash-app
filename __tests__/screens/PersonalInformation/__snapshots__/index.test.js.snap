// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonalInformation/> PersonalInformation renders correctly 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
